### PR TITLE
e2e tst: add actions to create github users

### PR DESCRIPTION
**Pull request by burmudar**
add actions to create github users and also remove created users.

note: we cannot create admins so instead of search and map a user to the
Admin user

## Test plan
tested in wb/e2e/poc

### CODE REVIEW

File: dev/tst/github_client.go [hunk 1]
```diff
@@ -68,3 +68,28 @@ func (gh *GitHubClient) orgUsers(ctx context.Context, org *github.Organization)
 	}
 	return users, nil
 }
+
+func (gh *GitHubClient) CreateUser(ctx context.Context, name, email string) (*github.User, error) {
+	user, resp, err := gh.c.Admin.CreateUser(ctx, name, email)
```
Comment (jhchabran) [inline dev/tst/github_client.go:73]: From the godocs: 

> When err is nil, resp always contains a non-nil resp.Body. Caller should close resp.Body when done reading from it. 

We need to close `resp.Body` regardless of the status code if there's no error

File: dev/tst/github_client.go [hunk 2]
```diff
@@ -68,3 +68,28 @@ func (gh *GitHubClient) orgUsers(ctx context.Context, org *github.Organization)
...
+	if resp.StatusCode >= 400 {
```
Comment (jhchabran) [inline dev/tst/github_client.go:74]: If there's an error on the transport layer, the underlying HTTP Get will return an error and a nil response, which will panic here. We can't skip checking `err`

File: dev/tst/users.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+var User2 GitHubScenarioUser = *NewGitHubScenarioUser("user2")
+var User3 GitHubScenarioUser = *NewGitHubScenarioUser("user3")
+var User4 GitHubScenarioUser = *NewGitHubScenarioUser("user4")
+var User5 GitHubScenarioUser = *NewGitHubScenarioUser("user5")
+var User6 GitHubScenarioUser = *NewGitHubScenarioUser("user6")
+var User7 GitHubScenarioUser = *NewGitHubScenarioUser("user7")
+var User8 GitHubScenarioUser = *NewGitHubScenarioUser("user8")
+var User9 GitHubScenarioUser = *NewGitHubScenarioUser("user9")
+var User10 GitHubScenarioUser = *NewGitHubScenarioUser("user10")
+var Admin GitHubScenarioUser = *NewGitHubScenarioUser("admin")
```
Comment (jhchabran) [inline dev/tst/users.go:46]: What is this about?

**Review by jhchabran — CHANGES_REQUESTED** [review]

File: dev/tst/users.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+func userEmail(u *GitHubScenarioUser) string {
+	return "tst-pkg-user@sourcegraph.com"
+}
+
+func (u *GitHubScenarioUser) GetUserAction(client *GitHubClient) Action {
+	name := u.Name()
+	if u.Name() == Admin.Name() {
+		name = client.cfg.User
```
Comment (burmudar) [inline dev/tst/users.go:96]: should be `client.cfg.AdminUser`

File: dev/tst/github_client.go (hunk 1 above)
Comment (burmudar) [reply dev/tst/github_client.go]: Fixed

File: dev/tst/users.go (hunk 4 above)
Comment (burmudar) [reply dev/tst/users.go]: Fixed

File: dev/tst/users.go (hunk 3 above)
Comment (burmudar) [reply dev/tst/users.go]: no longer relevant

**Review by jhchabran — APPROVED** [review]
pre-approving to unblock

File: dev/codehost_testing/scenario.go [hunk 5]
```diff
@@ -243,3 +243,47 @@ func (s *GithubScenario) CreateOrg(name string) *Org {
...
+	return baseUser
+}
+
+// GetAdmin returns a User representing the GitHub admin user configured in the client.
+//
+// NOTE: this method does not actually add an explicit action to the scenario, but will still
+// require that the scenario has been applied before the admin user can be retrieved - even though
+// it is not strictly required as the Admin already exists.
+func (s *GithubScenario) GetAdmin() *User {
+	baseUser := &User{
```
[the remark this answers is not in the document]
Comment (burmudar) [reply dev/codehost_testing/scenario.go]: Fixed

File: dev/codehost_testing/scenario.go [hunk 6]
```diff
@@ -243,3 +243,47 @@ func (s *GithubScenario) CreateOrg(name string) *Org {
...
+		s:    s,
+		name: name,
+	}
+
+	createUser := &action{
+		name: "user:create:" + name,
+		apply: func(ctx context.Context) error {
+			name := fmt.Sprintf("user-%s-%s", name, s.id)
+			email := "test-user-e2e@sourcegraph.com"
```
[the remark this answers is not in the document]
Comment (burmudar) [reply dev/codehost_testing/scenario.go]: Good catch - yes this caused an issue. It is fixed now. Tested by adding another user and adding a unique email id

**Pull request merged**